File: Lecture17: Bubble Sort/bubbleSort.py
def bubbleSort(arr, n):
	for i in range(1, n):

		for j in range(n-1):
			if arr[j] > arr[j+1]:
				arr[j], arr[j+1] = arr[j+1], arr[j]


if __name__ == '__main__':
	arr = [2, 5, 3, 6, 1, 7, 4]
	n = len(arr)

	print("Before sorting")
	print(arr)

	bubbleSort(arr, n)

	print("After sorting")
	print(arr)

File: Lecture017 Bubble Sort/BubbleSort.java


import java.util.Scanner;
import java.util.Arrays;
public class BubbleSort {
    // Bubble Sort -> Also Known as Sinking Sort and Exchange Sort
    public static void main(String [] args){
        /*
         */

        // Creating Scanner Object for taking inputs from the USER
        Scanner sc = new Scanner(System.in);
        // Take the SIZE of 1D Array from the USER
        System.out.print(" Enter SIZE of 1D Array: ");
        int n = sc.nextInt();
        // Create Array of Required SIZE
        int [] arr = new int[n];
        //Fill the Array by taking n Inputs from the USER
        System.out.print(" Enter " + n + " Elements: ");
        for(int i=0; i<arr.length; i++)
            arr[i] = sc.nextInt();
        // Display Array Before Sorting
        System.out.println(" Array Before Sorting is: " + Arrays.toString(arr));
        // Apply Bubble Sort on the Given Array
        bubbleSort(arr);
        // Display Array After Sorting
        System.out.println(" Array After Sorting is: " + Arrays.toString(arr));
    }
    // Method of Bubble Sort
    static void bubbleSort(int [] arr){
        for(int i=0; i<arr.length-1 ; i++){
            boolean isSorted = true;
            for(int j = 0; j<arr.length-i-1; j++){
                if(arr[j] > arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    isSorted = false;
                }
            }
            if(isSorted)
                break;

        }
    }
}
